modules/vre/spec/lib/vre/vre_monitor_spec.rb

```diff
@@ -199,45 +199,57 @@
-                   log_message += ' submission to LH begun'
-                   [base_payload.except(:attachments, :file), nil]
-                 when 'attempt'
-                   log_message += ' submission to LH attempted'
-                   [base_payload, nil]
-                 when 'success'
-                   log_message += ' submission to LH succeeded'
-                   [base_payload.except(:attachments, :file), nil]
-                 when 'failure'
-                   log_level = :warn
-                   log_message += ' submission to LH failed, retrying'
-                   [base_payload.except(:attachments, :file).merge(message: monitor_error.message), nil]
-                 when 'exhausted'
-                   log_level = :error
-                   log_message += ' submission to LH exhausted!'
-                   msg = { 'args' => [claim.id, current_user.user_account_uuid] }
-                   payload = if has_confirmation_number
-                               {
-                                 confirmation_number: claim.confirmation_number,
-                                 user_account_uuid: current_user.user_account_uuid,
-                                 form_id: claim.form_id,
-                                 claim_id: claim.id,
-                                 message: msg,
-                                 tags: monitor.tags
-                               }
-                             else
-                               {
-                                 confirmation_number: nil,
-                                 user_account_uuid: current_user.user_account_uuid,
-                                 form_id: nil,
-                                 claim_id: claim.id,
-                                 message: msg,
-                                 tags: monitor.tags
-                               }
-                             end
-                   [payload, msg]
-                 end
+      payload, msg = case monitor_event
+                     when 'begun'
+                       log_message += ' submission to LH begun'
+                       [base_payload.except(:attachments, :file), nil]
+                     when 'attempt'
+                       log_message += ' submission to LH attempted'
+                       [base_payload, nil]
+                     when 'success'
+                       log_message += ' submission to LH succeeded'
+                       [base_payload.except(:attachments, :file), nil]
+                     when 'failure'
+                       log_level = :warn
+                       log_message += ' submission to LH failed, retrying'
+                       [base_payload.except(:attachments, :file).merge(message: monitor_error.message), nil]
+                     when 'exhausted'
+                       log_level = :error
+                       log_message += ' submission to LH exhausted!'
+                       msg = { 'args' => [claim.id, current_user.user_account_uuid] }
```

Comment:
Very minor style nit, but why define msg here? Doing it here and returning it from the case statement means all the other cases have to return nil. Instead, you could define this in the `if monitor_event == 'exhausted'` block below. the `msg` variable will automatically be nil for all other cases. ruby defines variables and sets them to nil if they're defined in an un-executed condition:
```
vets-api(dev)* if 1 == 1
vets-api(dev)*   puts "do nothing"
vets-api(dev)* else
vets-api(dev)*   foo = "be nothing"
vets-api(dev)> end
do nothing
=> nil
vets-api(dev)> foo
=> nil
```